compose/ui/ui/src/commonMain/kotlin/androidx/compose/ui/keepscreenon/KeepScreenOnManager.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.ui.keepscreenon
+
+internal class KeepScreenOnManager {
+    val isKeepScreenOnEnabled: Boolean get() = keepScreenOnCount > 0
+    private val clientCounts = mutableMapOf<Any, Int>()
```

Comment:
Let's avoid singletons and such maps in shared code. 

Keep just
```
private var keepScreenOnCount = 0
```

inside `Owner` and report the bool status to `PlatformContext`. Combine global state on the platform side in case of need

Such a global object doesn't fit well to all platforms. See android implementation as a ref